src/Std/Time/Date/Unit/Month.lean

```diff
@@ -0,0 +1,190 @@
+Authors: Sofia Rodrigues
+-/
+prelude
+import Std.Time.UnitVal
+import Std.Time.Bounded
+import Std.Time.LessEq
+import Lean.Data.Rat
+import Std.Time.Time.Basic
+import Std.Time.Date.Unit.Day
+
+namespace Std
+namespace Time
+open Lean Time
+
+namespace Month
+/--
+`Ordinal` represents a bounded value for months, which ranges between 1 and 12.
+-/
+def Ordinal := Bounded.LE 1 12
+  deriving Repr, BEq, LE
+
+instance [Le 1 n] [Le n 12] : OfNat Ordinal n where
+  ofNat := Bounded.LE.mk (Int.ofNat n) (And.intro (Int.ofNat_le.mpr Le.p) (Int.ofNat_le.mpr Le.p))
+
+instance : Inhabited Ordinal where default := 1
+
+/--
+`Offset` represents an offset in months. It is defined as an `Int`.
+-/
+def Offset : Type := Int
+  deriving Repr, BEq, Inhabited, Add, Sub, Mul, Div, Neg, ToString
+
+instance : OfNat Offset n := ⟨Int.ofNat n⟩
+
+namespace Ordinal
+
+@[inline] def january : Ordinal := 1
+@[inline] def february : Ordinal := 2
+@[inline] def march : Ordinal := 3
+@[inline] def april : Ordinal := 4
+@[inline] def may : Ordinal := 5
+@[inline] def june : Ordinal := 6
+@[inline] def july : Ordinal := 7
+@[inline] def august : Ordinal := 8
+@[inline] def september : Ordinal := 9
+@[inline] def october : Ordinal := 10
+@[inline] def november : Ordinal := 11
+@[inline] def december : Ordinal := 12
+
+/--
+Creates an `Ordinal` from a natural number, ensuring the value is within bounds.
+-/
+@[inline]
+def ofNat (data : Nat) (h : data ≥ 1 ∧ data ≤ 12 := by decide) : Ordinal :=
+  Bounded.LE.ofNat' data h
+
+/--
+Converts a `Ordinal` into a `Nat`.
+-/
+@[inline]
+def toNat (month : Ordinal) : Nat :=
+  Bounded.LE.toNat month
+
+/--
+Creates an `Ordinal` from a `Fin`, ensuring the value is within bounds, if its 0 then its converted
+to 1.
+-/
+@[inline]
+def ofFin (data : Fin 13) : Ordinal :=
+  Bounded.LE.ofFin' data (by decide)
+
+/--
+Transforms `Month.Ordinal` into `Second.Offset`.
+-/
+def toSeconds (leap : Bool) (month : Ordinal) : Second.Offset :=
+  let daysAcc := #[0, 31, 59, 90, 120, 151, 181, 212, 243, 273, 304, 334]
+  let time := daysAcc[month.toNat]! * 86400
+  if leap && month.toNat ≥ 2
+    then time + 86400
+    else time
+
+/--
+Transforms `Month.Ordinal` into `Minute.Offset`.
+-/
+@[inline]
+def toMinute (leap : Bool) (month : Ordinal) : Minute.Offset :=
+  toSeconds leap month
+  |>.div 60
+
+/--
+Transforms `Month.Ordinal` into `Hour.Offset`.
+-/
+@[inline]
+def toHours (leap : Bool) (month : Ordinal) : Hour.Offset :=
+  toMinute leap month
+  |>.div 60
+
+/--
+Transforms `Month.Ordinal` into `Day.Offset`.
+-/
+@[inline]
+def toDays (leap : Bool) (month : Ordinal) : Day.Offset :=
+  toSeconds leap month
+  |>.convert
+
+/--
+Size in days of each month if the year is not leap.
+-/
+@[inline]
+def monthSizesNonLeap : { val : Array Day.Ordinal // val.size = 12 } :=
+  ⟨#[ 31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31 ], by simp⟩
+
+/--
+Gets the number of days in a month.
+-/
+def days' (leap : Bool) (month : Ordinal) : Day.Ordinal :=
```

Comment:
It would be great if we could a naming scheme for functions that are part of the public API that does not rely on `'` to disambiguate functions, since these names are very difficult to guess.